### ISSUE TITLE
Incorrect whitespace enforced before questionmark with omitted type

### ISSUE THREAD

**Issue by sebek64** [palantir/tslint#3916]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.5.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class X {
  private a? = false;
  private b?: string[] = [];
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "whitespace": {"options": ["check-branch", "check-decl", "check-operator", "check-separator", "check-type", "check-typecast"]
    }
  }
}
```

#### Actual behavior

Reports error on member a, but no error on member b. Fixer corrects it by adding a whitespace before ?.

#### Expected behavior

This code should be valid and no error should be reported, in my opinion.

**Comment by fedorl** [comment]
It seems to be broken since 5.9.0, we had to rollback to 5.8.0. This issue blocks further tslint upgrades (unless one wants to reformat existing code to this weird spacing or disable the rule completely).

**Comment by giladgray** [comment]
I would argue that `private a? = false` with that default assignment should just be `private a = false`.

but yes i can repro this.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
